include/BatchedContext.hpp

```diff
@@ -584,7 +584,12 @@ class BatchedContext
     }
     void TRANSLATION_API ReleaseResource(Resource* pResource)
     {
+        auto Size = pResource->GetResourceSize();
         AddPostBatchFunction([pResource]() { pResource->Release(); });
+        if (Size >= 32 * 1024 * 1024)

```

Comment:
Possible to add a running total of size of pending resource releases to compare against instead?

Reply:
Done. I bumped to 64MB pending destruction. That seems like a reasonable amount to me.